pkg/policy/resolve.go

```diff
@@ -267,11 +270,12 @@ type PolicyOwner interface {
 }
 
 // newSelectorPolicy returns an empty selectorPolicy stub.
-func newSelectorPolicy(selectorCache *SelectorCache) *selectorPolicy {
+func newSelectorPolicy(selectorCache *SelectorCache, bakery cookie.PolicyBakery) *selectorPolicy {
```

Comment:
Hm, why does each selector policy need a reference to the bakery? Could you instead pass it to `DistillPolicy`?